atd-toolbox/cris_export_parser/parse_data.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+
+import csv
+import argparse
+import psycopg2
+import psycopg2.extras
+import time
+
+conn = psycopg2.connect(
+    dbname="atd_vz_data",
+    user="visionzero",
+    password="visionzero",
+    host="localhost",
+    port="5432",
+)
+
+
+def parse_csv(file_path, field_names, substrings):
+    with open(file_path, "r") as file:
+        reader = csv.DictReader(file)
+        for row in reader:
+            for field in field_names:
+                # Check if field is in row and not None
+                if field in row and row[field] is not None:
+                    fieldValue = row[field]  # Store the field value
+                    for substring in substrings:
+                        if substring.lower() in fieldValue.lower():
+                            if not (
+                                "Crash_ID" in row
+                                and "Unit_Nbr" in row
+                                and "Prsn_Nbr" in row
+                            ):
+                                continue
+
+                            with conn.cursor(
+                                cursor_factory=psycopg2.extras.RealDictCursor
+                            ) as cur:
+                                # Use the cursor here
+
+                                count = f"""
+select count(*) as count
+from atd_txdot_primaryperson
+where true
+and crash_id = {row.get("Crash_ID")} 
+and unit_nbr = {row.get("Unit_Nbr")}
+and prsn_nbr = {row.get("Prsn_Nbr")};"""
+
+                                cur.execute(count)
+                                count_check = cur.fetchone()
```

Comment:
cool, so this checks that a record match exists before trying to update its `peh_fl`and logs it if there isn't a match and is skipped 👍